ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -828,26 +832,30 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 // (1) shouldDelete: Whether the Pod should be deleted.
 // (2) reason: The reason why the Pod should or should not be deleted.
 func shouldDeletePod(pod corev1.Pod, nodeType rayv1.RayNodeType) (bool, string) {
-	// If a Pod's restart policy is set to `Always`, KubeRay will not delete
+	// If a Pod's restart policy is set to `Always` and Pod not be evicted, KubeRay will not delete
 	// the Pod and rely on the Pod's restart policy to restart the Pod.
 	isRestartPolicyAlways := pod.Spec.RestartPolicy == corev1.RestartPolicyAlways
+	isPodEvicted := pod.Status.Reason == podEvictedReason
 
 	// If the Pod's status is `Failed` or `Succeeded`, the Pod will not restart and we can safely delete it.
 	if pod.Status.Phase == corev1.PodFailed || pod.Status.Phase == corev1.PodSucceeded {
-		if isRestartPolicyAlways {
-			// Based on my observation, a Pod with `RestartPolicy: Always` will never be in the terminated states (i.e., `Failed` or `Succeeded`).
-			// However, I couldn't find any well-defined behavior in the Kubernetes documentation, so I can't guarantee that the status transition
-			// from `Running` to `Failed / Succeeded` and back to `Running` won't occur when we kill the main process (i.e., `ray start` in KubeRay)
-			// in the head Pod. Therefore, I've added this check as a safeguard.
+		if isRestartPolicyAlways && !isPodEvicted {
+			// A Pod with `RestartPolicy: Always` will guarantee that the status transition
+			// from `Running` to `Failed / Succeeded` and back to `Running`.
+			// However, if the Non-Controller Managed Pod(e.g., head pod) is evicted
```

Comment:
what does non-controller managed Pod mean?

Reply:
> what does non-controller managed Pod mean?

"non-controller managed Pod" refers to Pods that are directly created by users without being managed by any Kubernetes controllers (such as Deployments, ReplicaSets, StatefulSets, etc.). 
For example, If pod is managed by ReplicaSets, when it is evicted, it will be re-scheduled